image/manifest.go

```diff
@@ -89,7 +89,7 @@ func (m *manifest) validate(w walker) error {
 
 func (m *manifest) unpack(w walker, dest string) error {
 	for _, d := range m.Layers {
-		if d.MediaType != string(schema.MediaTypeImageConfig) {
+		if d.MediaType != string(schema.MediaTypeImageLayer) {
 			continue
```

Comment:
On Wed, Sep 07, 2016 at 04:58:52AM -0700, Lei Jitang wrote:
> > @@ -89,7 +89,7 @@ func (m *manifest) validate(w walker) error {
> >  
> >  func (m *manifest) unpack(w walker, dest string) error {
> >  	for _, d := range m.Layers {
> > -		if d.MediaType != string(schema.MediaTypeImageConfig) {
> > +		if d.MediaType != string(schema.MediaTypeImageLayer) {
> >  			continue
>
> @wking if the layers MUST have a particular type is not said in the
> spec currently, I think we should make it clear in the spec first.

If the layers MUST have a particular type, then you should error out
here with “your manifest is broken”.  If the spec does not limit the
allowed types (like now), then you should error out here with
“unimplemented type”.  In no case that I can see should we be silently
ignoring an unrecognized type.